utils/average.go

```diff
@@ -7,8 +7,8 @@ import "sync/atomic"
 // used for statistical purposes only due to the use of
 // atomics and not mutexes.
 type RollingAverage struct {
-	data []int64
```

Comment:
im worried about trusting alignment for this. can you see what happens if you use an atomic.Int64 without changing the ordering?